### PR TITLE
Prompt Registry run

### PR DESCRIPTION
**Pull request by takeshicamilo**
With this update basically I added a new function that let the user run their registries and test it in the way they want

promptlayer.prompts.run("runRegistry", False, ['funciona?', 'xD'], 2, 'text-ada-001')

and you need to use your promptLayer API and openai or anthropic
It need 6 parameters
- name of registry 
- lang (can be true or false) 
- tags,
- version
- engine
- model (openai or anthropic)

### CODE REVIEW

File: promptlayer/__init__.py [hunk 1]
```diff
@@ -12,12 +12,10 @@
 def __getattr__(name: Union[Literal["openai"], Literal["anthropic"]]):
     if name == "openai":
         import openai as openai_module
-
```
Comment (Jped) [inline promptlayer/__init__.py:15]: There are a lot of random additional spaces in this file. You might have used an automatic formatter or something. We use black (you can instally it on pypi).

Make sure when you have a PR only relevant changes show up and not spacing things like this.

File: promptlayer/prompts/prompts.py [hunk 2]
```diff
@@ -25,6 +25,17 @@ def get_prompt(prompt_name, langchain=False, version=None):
...
+    prompt = get_prompt(prompt_name, langchain=False, version=None)
+    obj = {
+        'prompt_name': prompt_name,
+        'version': version,
+        'tags': tags,
+        'engine': engine,
+        'model': model
+    }
+    # Get the current time in seconds since the epoch
+    run_prompt_registry(prompt, obj)
```
Comment (Jped) [inline promptlayer/prompts/prompts.py:38]: why are you passing this in as an object?

This is inconsistent with how other functions are called in this repo. We should stay consistent.

Comment (Jped) [inline promptlayer/prompts/prompts.py:38]: we should return the pl_id from this function

File: promptlayer/utils.py [hunk 3]
```diff
@@ -24,6 +25,107 @@ def get_api_key():
     else:
         return promptlayer.api_key
 
+def run_prompt_registry(prompt, obj):
+
+    current_time = time.time()
+    print(obj)
```
Comment (Jped) [inline promptlayer/utils.py:31]: print statements and comments should not be in a PR, there are a few others all around. Please remove them

File: promptlayer/utils.py [hunk 4]
```diff
@@ -24,6 +25,107 @@ def get_api_key():
...
+                        "role": "user"
+                    }
+                    ]},
+                    "tags": obj['tags'],
+                    "request_response": completion,
+                    "request_start_time": request_start_time,
+                    "request_end_time": request_end_time,
+                    # "prompt_id": prompt_name,
+                    "prompt_input_variables": "",
+                    "prompt_version": obj['version'],
```
Comment (Jped) [inline promptlayer/utils.py:72]: Why are you not setting the `prompt_input_variables` and the `prompt_id` here?

File: promptlayer/utils.py [hunk 5]
```diff
@@ -24,6 +25,107 @@ def get_api_key():
...
+
+            #generate a new Completion prompt
+            completion = openai.Completion.create(
+                model=obj['engine'],
+                prompt=message,
+                max_tokens=7,
+                temperature=0
+            )
+            #generate a new track with the answer of the completion
+            request_response = requests.post(
```
Comment (Jped) [inline promptlayer/utils.py:91]: This is essentially a copy of the request in the above if statement. Why is it in the if statement? 

You should not repeat code blocks.

File: run.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import promptlayer
```
Comment (Jped) [inline run.py:1]: why is this file included in the PR?

File: promptlayer/__init__.py (hunk 1 above)
Comment (Jped) [reply promptlayer/__init__.py:15]: @LilSherlock there is still spaces in this __init__.py file

File: promptlayer/utils.py [hunk 7]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+        )
+        # Get the id of the request
+        requestID = request_response.json()["request_id"]
+    else:
+        if "messages" in prompt:
+            message = prompt["messages"][1]["prompt"]["template"]
+        else:
+            message = prompt["template"]
+
+        anthropic = promptlayer.anthropic
```
Comment (Jped) [inline promptlayer/utils.py:102]: Why are you using the promptlayer wrapper for anthropic but for the OpenAI requests you are using OpenAI directly?

File: promptlayer/utils.py [hunk 8]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+
+    if model == "openai":
+        if "messages" in prompt:
+            # Get the system message
+            template = prompt["messages"][0]["prompt"]["template"]
+            # Get user message
+            message = prompt["messages"][1]["prompt"]["template"]
+
+            # Generate new ChatCompletionPrompt
+            completion = openai.ChatCompletion.create(
```
Comment (Jped) [inline promptlayer/utils.py:51]: Another thing to consider is that neither `openai` or `anthropic` are required libraries in PromptLayer. So it can be that someone is trying to use the `promptlayer.prompts.run` and we dont have this installed.

What would be the error? We should try to catch that here and have good error support

File: promptlayer/utils.py [hunk 9]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+                model=engine,
+                messages=[
+                    {"role": "system", "content": template},
+                    {"role": "user", "content": message},
+                ],
+            )
+            # print(completion)
```
Comment (Jped) [inline promptlayer/utils.py:58]: remove this print statement

File: promptlayer/utils.py [hunk 10]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+
+        response = anthropic_client.completion(
+            prompt=f"{anthropic.HUMAN_PROMPT} {message}{anthropic.AI_PROMPT}",
+            stop_sequences=[anthropic.HUMAN_PROMPT],
+            model="claude-v1-100k",
+            max_tokens_to_sample=100,
+            pl_tags=tags,
+            return_pl_id=True,
+        )
+        return
```
Comment (Jped) [inline promptlayer/utils.py:114]: why are we not returning the pl_id here?

File: promptlayer/utils.py [hunk 11]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+    response = requests.get(
+        "https://api.promptlayer.com/rest/get-prompt-template",
+        headers={"X-API-KEY": api_key},
+        params={"prompt_name": prompt_name, "version": version},
+    )
+    prompt = response.json()["prompt_template"]
+    prompt_id = response.json()["id"]
+    # Update the request_start_time and request_end_time with the current time
+    request_start_time = current_time
+    request_end_time = current_time
```
Comment (Jped) [inline promptlayer/utils.py:41]: why are these the same time?

File: promptlayer/utils.py [hunk 12]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+
+            # Generate a new Completion prompt
+            completion = openai.Completion.create(
+                model=engine, prompt=message, max_tokens=7, temperature=0
+            )
+            function_name = "openai.Completion.create"
+            kwargs = {"engine": engine, "prompt": message}
+
+        # Track the request
+        request_response = requests.post(
```
Comment (Jped) [inline promptlayer/utils.py:79]: use the utilty function in utils.py. You will need to edit it to take `prompt_id` `prompt_input_variables` and `prompt_version`

File: promptlayer/utils.py [hunk 13]
```diff
@@ -25,6 +26,94 @@ def get_api_key():
...
+                "prompt_id": prompt_id,
+                "request_start_time": request_start_time,
+                "request_end_time": request_end_time,
+                "prompt_input_variables": variables,
+                "prompt_version": version,
+                "api_key": promptlayer.api_key,
+            },
+        )
+        # Get the id of the request
+        requestID = request_response.json()["request_id"]
```
Comment (Jped) [inline promptlayer/utils.py:95]: `pl_request_id`

File: promptlayer/utils.py (hunk 10 above)
Comment (Jped) [reply promptlayer/utils.py:114]: `pl_request_id`

File: promptlayer/utils.py [hunk 14]
```diff
@@ -5,9 +5,10 @@
 import sys
 import types
 from copy import deepcopy
-
 import requests
-
+import time
+import openai
```
Comment (Jped) [inline promptlayer/utils.py:10]: remove `import openai` and `import anthropic` from the top here and move them down to the function below

File: promptlayer/utils.py [hunk 15]
```diff
@@ -330,7 +333,7 @@ def promptlayer_track_prompt(
             json={
                 "request_id": request_id,
                 "prompt_name": prompt_name,
-                "prompt_input_variables": input_variables,
+                # "prompt_input_variables": input_variables,
```
Comment (LilSherlock) [inline promptlayer/utils.py:336]: I had to comment this because I was getting an error about that promp_input_variables has to be a json

File: promptlayer/utils.py [hunk 16]
```diff
@@ -76,24 +68,28 @@ def run_prompt_registry(prompt_name, version, tags, variables, engine, model, ap
...
-                "prompt_input_variables": variables,
-                "prompt_version": version,
-                "api_key": promptlayer.api_key,
-            },
+        current_time = time.time()
+        request_end_time = current_time
+
+        request_id = promptlayer_api_request(
+            function_name,
+            "langchain",  # ask what is this
```
[the remark this answers is not in the document]
Comment (Jped) [reply promptlayer/utils.py:76]: provider type for openai should be 'openai'

File: promptlayer/utils.py [hunk 17]
```diff
@@ -25,6 +24,95 @@ def get_api_key():
...
+            variables,
+            kwargs,
+            tags,
+            completion,
+            request_start_time,
+            request_end_time,
+            get_api_key(),
+            True,
```
Comment (Jped) [inline promptlayer/utils.py:84]: these last for arguments `True`, `None`, `prompt_id`, and `version` are all optional arguments. 
Skip the `True` and `None` and pass the last two with their key ie

`prompt_id=prompt_id` and `prompt_version=version`

File: promptlayer/utils.py [hunk 18]
```diff
@@ -5,9 +5,8 @@
 import sys
 import types
 from copy import deepcopy
-
 import requests
-
+import time
```
Comment (Jped) [inline promptlayer/utils.py:9]: we use `datetime` elsewhere in this library. Lets use that instead of `time` to keep things consistent.

File: promptlayer/utils.py [hunk 19]
```diff
@@ -25,6 +24,95 @@ def get_api_key():
         return promptlayer.api_key
 
 
+def run_prompt_registry(prompt_name, version, tags, variables, engine, model, api_key):
+    response = promptlayer_get_prompt(prompt_name, api_key, version)
+    prompt = response["prompt_template"]
+    prompt_id = response["id"]
+    # Update the request_start_time and request_end_time with the current time
+    current_time = time.time()
```
Comment (Jped) [inline promptlayer/utils.py:32]: This line and the next time can be merged.

`request_start_time = datetime.datetime.now().timestamp()`

File: promptlayer/utils.py [hunk 20]
```diff
@@ -25,6 +24,95 @@ def get_api_key():
         return promptlayer.api_key
 
 
+def run_prompt_registry(prompt_name, version, tags, variables, engine, model, api_key):
+    response = promptlayer_get_prompt(prompt_name, api_key, version)
```
Comment (Jped) [inline promptlayer/utils.py:28]: you use the variable name `response` for something else below, rename this to something better
